packages/flutter_secure_storage/tizen/src/flutter_secure_storage.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter_secure_storage.h"
+
+#include <ckmc/ckmc-manager.h>
+#include <ckmc/ckmc-type.h>
+
+#include <algorithm>
+
+#include "log.h"
+#include "secure_storage_util.h"
+
+FlutterSecureStorage::FlutterSecureStorage() {
```

Comment:
> Why are these classes named Cipher and FlutterSecureStorage? Should they be separate classes? They seem to be both a wrapper around the same "ckmc" module.

I thought it would be better if their roles were separated.

>Especially you should not use the Flutter prefix in FlutterSecureStorage since the class is not related to Flutter. Also consider changing the return type of the ReadAll method to std::vector.

OK, I'll update

Reply:
> Also consider changing the return type of the ReadAll method to std::vector.

Did you have this in mind?
```C++
  std::vector<std::pair<std::string, std::string>> 
```

Reply:
Ah, it's `std::map`.

I still think `Cipher` and `FlutterSecureStorage` should be a single class or at least the code should be refactored so that `SecureStorageUtil` becomes part of `FlutterSecureStorage`.

Reply:
I thought this was better in terms of SRP... but alright, no problem. I'll move them all to `SecureStorage`

Reply:
> Ah, it's std::map

@swift-kim 
If I apply this, `secure_storage` is no longer dependent on the `flutter::EncodableMap` but I need to convert to `flutter::EncodableMap `too in caller side to send result. This seems a bit inefficient, what do you think?

```c++
std::map<std::string, std::string> SecureStorage::ReadAll() {
  std::vector<std::string> keys =
      SecureStorageUtil::GetAliasList(AliasType::kData);

  std::map<std::string, std::string> key_value_pairs;
  for (std::string key : keys) {
    std::optional<std::string> value = Read(key);
    key_value_pairs[key] = value.value();
  }

  return key_value_pairs;
}
```

```c++
      std::map<std::string, std::string> map = storage_.ReadAll();
      flutter::EncodableMap encodable_map;
      for (auto iter : map) {
        encodable_map[flutter::EncodableValue(iter.first)] =
            flutter::EncodableValue(iter.second);
      }
      result->Success(flutter::EncodableValue(encodable_map));
```

Reply:
We often to that in other plugins, e.g.

https://github.com/flutter-tizen/plugins/blob/ca3d8b9e49304c4bdb272f20ec6474dc49428e6e/packages/tizen_app_manager/tizen/src/tizen_app_manager_plugin.cc#L173-L176

Reply:
Alright and your suggestion is more modern expression:+1: